packages/json-spell-checker/index.js

```diff
@@ -0,0 +1,402 @@
+const jsonMap = require('json-source-map')
+
+module.exports.configure = configure
+module.exports.checkTypos = checkTypos 
+
+const REF = "$ref"
+const PROPERTIES = "properties"
+const PATTERN_PROPERTIES = "patternProperties"
+const ADDITONAL_PROPERTIES = "additionalProperties"
+const PATH = "#/"
+
+let SCHEMA = require('./examples/schema/td-schema.json')
+// Minimum similarity value to be able to say that two words are similar
+let SIMILARITY_THRESHOLD = 0.85
+// Maximum value of length difference between two words
+let MAX_LENGTH_DIFFERENCE = 2
+let TYPO_LOOKUP_TABLE = createSchemaLookupTable(SCHEMA)
+
+
+/**
+ * Configures the settings of the spell checker
+ * @param {*} similarityThreshold Threshold that decides whether the words are similar or not. 
+ * @param {*} maxLengthDifference Maximum value of the length difference of words to run the algorithm
+ */
+function configure(schema = SCHEMA, similarityThreshold = SIMILARITY_THRESHOLD, maxLengthDifference = MAX_LENGTH_DIFFERENCE) {
+    SCHEMA = schema
+    TYPO_LOOKUP_TABLE = createSchemaLookupTable(SCHEMA)
+    SIMILARITY_THRESHOLD = similarityThreshold
+    MAX_LENGTH_DIFFERENCE = maxLengthDifference
+}
+
+/**
+ * Checks possible typos in a TD
+ * @param {object} td The TD to apply typo check on
+ * @returns List of possible typos where the typo consists of string value of typo itself and the message,
+ * another string value, to be prompted to the user for the fix
+ */
+function checkTypos(td) {
+    const typos = []
+
+    const lookupTable = TYPO_LOOKUP_TABLE
+    const searchDepth = 1
+    const searchPath = PATH
+    let tdJson = {}
```

Comment:
if code uses td, it should also be just json

Reply:
[a9f2bdd](https://github.com/thingweb/thingweb-playground/pull/443/commits/a9f2bdd96a2a0fc3404919c280cec59291109d2c)